espresso-support/core/src/main/java/com/novoda/espresso/AccessibilityViewMatchers.java

```diff
@@ -0,0 +1,187 @@
+
+                    return expectedText.equals(clickAction.getLabel());
+                } else {
+                    return false;
+                }
+            }
+
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("is long clickable and has custom usage hint for ACTION_LONG_CLICK from resource id: ");
+                description.appendValue(resourceId);
+                if (null != this.resourceName) {
+                    description.appendText("[");
+                    description.appendText(resourceName);
+                    description.appendText("]");
+                }
+                if (null != this.expectedText) {
+                    description.appendText(" value: ");
+                    description.appendText(expectedText);
+                }
+            }
+        };
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    public static Matcher<? super View> withUsageHintOnLongClick(CharSequence text) {
+        return withUsageHintOnLongClick(is(text));
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    public static Matcher<? super View> withUsageHintOnLongClick(final Matcher<? extends CharSequence> charSequenceMatcher) {
+        return new TypeSafeMatcher<View>() {
+            @Override
+            protected boolean matchesSafely(View view) {
+                if (!view.isLongClickable()) {
+                    return false;
+                }
+
+                AccessibilityNodeInfo accessibilityNodeInfo = view.createAccessibilityNodeInfo();
+                AccessibilityNodeInfo.AccessibilityAction clickAction = findAction(accessibilityNodeInfo, AccessibilityNodeInfo.ACTION_LONG_CLICK);
+
+                return charSequenceMatcher.matches(clickAction.getLabel());
+            }
+
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("is long clickable and has custom usage hint for ACTION_LONG_CLICK: ");
+                charSequenceMatcher.describeTo(description);
+            }
+        };
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    public static AccessibilityNodeInfo.AccessibilityAction findAction(AccessibilityNodeInfo accessibilityNodeInfo, int actionId) {
+        for (AccessibilityNodeInfo.AccessibilityAction accessibilityAction : accessibilityNodeInfo.getActionList()) {
+            if (actionId == accessibilityAction.getId()) {
+                return accessibilityAction;
+            }
+        }
+        throw new AccessibilityActionNotFoundException(actionId);
+    }
+
+    public static class AccessibilityActionNotFoundException extends RuntimeException {
+
+        public AccessibilityActionNotFoundException(int actionId) {
+            super("Could not find AccessibilityAction with id: " + actionId);
+        }
+    }
```

Comment:
commit 253db0766c1bebd86ca112ed8b717833a4280c51
Author: Ataul Munim <hello@ataulm.com>
Date:   Tue Mar 28 13:31:06 2017 +0100

    Add missing blank line before end of class